metis-dereference/metis-dereference-import/src/main/java/eu/europeana/metis/dereference/vocimport/utils/NonCollidingPathVocabularyTrie.java

```diff
@@ -0,0 +1,93 @@
+package eu.europeana.metis.dereference.vocimport.utils;
+
+import eu.europeana.metis.dereference.vocimport.exception.VocabularyImportException;
+import eu.europeana.metis.dereference.vocimport.model.Vocabulary;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+
+/**
+ * <p>This class implements a Trie with vocabularies categorized by path (so the same vocabulary
+ * can be present multiple times if it has multiple paths). When adding a new vocabulary, it checks
+ * whether there is already a path in the trie that collides with one of the paths of the the new
+ * vocabulary (either because a new path is a substring of an existing one or vice versa).</p>
+ * <p>Implementation note: This Trie is implemented in a traditional way. It has a root node
+ * representing the empty string. Since we don't allow collisions, we can only have vocabularies in
+ * the leaf nodes (that don't have children). The exception is the root node, and then only when the
+ * Trie is empty.</p>
+ */
+public class NonCollidingPathVocabularyTrie {
+
+  private final TrieNode rootNode = new TrieNode();
+
+  /**
+   * This method inserts a new vocabulary. It will add it by all its paths.
+   *
+   * @param vocabulary The vocabulary to insert.
+   * @throws VocabularyImportException In case one of the paths collides with one that's already
+   * present (or, possibly, with another path of the new vocabulary).
+   */
+  public void insert(Vocabulary vocabulary) throws VocabularyImportException {
+    for (String path : vocabulary.getPaths()) {
```

Comment:
Yes, but it will be order n^2. Whereas this solution does the comparison in order n. May not be too important now, but perhaps, when there are more vocabularies...

Reply:
Sure, but I think it's an overkill for this case..
I mean how many vocabularies would you have, that number would never exceed a threshold that would give us a problem I believe..
I understand the exciting part of creating such an algorithm, but we are overdoing it here for something so simple I think..
Yours thoughts?

Reply:
Okay since I feel some hesitation here, let's keep the trie implementation
Can you though provide some more documentation on the files, in the form of examples perhaps?

Reply:
Because I haven't replied yet? I'm sorry, I was working on something else... but you're right, I was not sure whether it is a good idea to remove a demonstrably better solution that is fully unit tested and replace it with something worse/less optimal... 

At least I didn't try to make it generic!

Anyway, I can definitely add a bit more documentation